proxy.go

```diff
@@ -95,9 +101,21 @@ func NewProxyFromConfig(conf ProxyConfig) (p Proxy, err error) {
 
 	// We need a convenient way to know if we're even using Consul later
 	if p.ConsulForwardService != "" || p.ConsulTraceService != "" {
+		log.Info("Using consul")
 		p.usingConsul = true
 	}
 
+	// check if we are running on Kubernetes
+	if _, err := os.Stat("/var/run/secrets/kubernetes.io/serviceaccount"); !os.IsNotExist(err) {
+		log.Info("Using Kubernetes")
+		p.usingKubernetes = true
+
+		//TODO don't overload this
+		if conf.ConsulForwardServiceName != "" {
```

Comment:
I think you can just remove this `if`? If it's a proxy it's only job is to forward, so it should be fine if we've determined it's on kubernetes.

Reply:
Reping on this?

Reply:
The name here is confusing, because we're overloading it. Basically, the `if` statement ensures that the process will exit with non-zero status if it's running on Kubernetes but no forwarding service has been configured. We want it to fail fast, or else it'll just start accumulating metrics from the local veneurs, and never end up flushing them.


I want to fix the name (hence the TODO), but that'll require some additional refactoring, and I'd like to merge this in first so we can iterate faster.